apps/android/app/src/main/java/chat/simplex/app/model/ChatModel.kt

```diff
@@ -211,27 +211,34 @@ class ChatModel(val controller: ChatController) {
     }
   }
 
-  fun markChatItemsRead(cInfo: ChatInfo) {
-    // update preview
-    val chatIdx = getChatIndex(cInfo.id)
-    if (chatIdx >= 0) {
-      val chat = chats[chatIdx]
-      val lastId = chat.chatItems.lastOrNull()?.id
-      if (lastId != null) {
-        chats[chatIdx] = chat.copy(chatStats = chat.chatStats.copy(unreadCount = 0, minUnreadItemId = lastId + 1))
-      }
-    }
+  fun markChatItemsRead(cInfo: ChatInfo, range: CC.ItemRange? = null) {
     // update current chat
+    var markedRead = 0
     if (chatId.value == cInfo.id) {
       var i = 0
       while (i < chatItems.count()) {
         val item = chatItems[i]
-        if (item.meta.itemStatus is CIStatus.RcvNew) {
+        if (item.meta.itemStatus is CIStatus.RcvNew && (range == null || (range.from..range.to).contains(item.id))) {
```

Comment:
```suggestion
        if (item.meta.itemStatus is CIStatus.RcvNew && (range == null || (range.from > item.id && item.id < range.to)) {
```

or, if the range should be inclusive replace with >=, <= - there is no reason to construct the range here.